backend/src/services/ApplicationService.js

```diff
@@ -811,28 +811,33 @@ export default class ApplicationService extends BasePocketService {
    * @async
    */
   async updateApplication(applicationId, applicationData) {
+
     if (PocketApplication.validate(applicationData)) {
       if (!await this.userService.userExists(applicationData.user)) {
         throw new DashboardError("User does not exists");
       }
 
-      const application = PocketApplication.createPocketApplication(applicationData);
       const filter = {
         "_id": ObjectID(applicationId)
       };
 
-      const applicationDB = await this.__decryptApplicationFields(await this.persistenceService.getEntityByFilter(APPLICATION_COLLECTION_NAME, filter));
+      const application = await this.__decryptApplicationFields(await this.persistenceService.getEntityByFilter(APPLICATION_COLLECTION_NAME, filter));
 
-      if (!applicationDB) {
+      if (!application) {
         throw new DashboardError("Application does not exists");
       }
 
-      const applicationToEdit = {
-        ...application,
-        publicPocketAccount: applicationDB.publicPocketAccount
-      };
-
-      applicationToEdit.id = applicationId;
+      const applicationToEdit = application;
+
+      // Update the new fields
+      applicationToEdit.name = application.name;
+      applicationToEdit.contactEmail = application.contactEmail;
+      applicationToEdit.description = application.description;
+      applicationToEdit.owner = application.owner;
+      applicationToEdit.url = application.url;
+      applicationToEdit.icon = application.icon;
+      applicationToEdit.user = application.user;
```

Comment:
I don't think we should permit the API to update the user, ever.  This would also close another issue I opened this morning.